pytket/extensions/qujax/qujax_convert.py

```diff
@@ -35,22 +36,24 @@ def _tk_qubits_to_inds(tk_qubits: Sequence[Qubit]) -> Tuple[int, ...]:
     return tuple(q.index[0] for q in tk_qubits)
 
 
-def tk_to_qujax(circuit: Circuit) -> UnionCallableOptionalArray:
+def tk_to_qujax_args(
+    circuit: Circuit,
+) -> Tuple[Sequence[str], Sequence[Sequence[int]], Sequence[Sequence[int]], int]:
     """
-    Converts a tket circuit into a function that maps circuit parameters
-    to a statetensor. Assumes all circuit gates can be found in qujax.gates.
+    Converts a tket circuit into a tuple of arguments to be sent to qujax.
```

Comment:
What are this arguments? I think the name of the function is a bit unintuitive, I would expect additional argument that needs to be passed in, maybe the name could be changed to `_expended` or so, to make that more clear?

Reply:
Have you thought about putting this all in the same function and changing the return value based on an additional parameter?

Reply:
The "qujax_args" are (gate_seq, qubit_inds_seq, param_inds_seq, n_qubits) which define a circuit in qujax (there is no circuit class in qujax). I think the name is ok but I will make this clearer in the docstring.

Reply:
Removing all of the *_symbolic functions and just having it inferred based on would be nice, and would clean up the code somewhat.

Currently we do not do this because the behaviour of `tk_to_qujax(circuit)` and `tk_to_qujax_symbolic(circuit, symbol_map=None)` is slightly different (in how it interprets parameterised gates) as described in the docstring of tk_to_qujax_symbolic.

As this change changes the API in a way that is not backwards compatible (it will break code that uses tk_to_qujax_symbolic), I suggest we leave this to a later release?

Reply:
I would not be worried about breaking the API in this point, doing it now or later is both fine with me.

Reply:
Have you thought about defining a qujax.circuit containing all this information and making this a bit more intuitive for the users? (No need to do that now, but might be good to do that some time in the future?)

Reply:
I would like to keep qujax without a qujax.circuit, JAX is built for functional programming (easier to debug, easier to compose with other code), so adding a custom class would go against this

So I think the workflow with the extension is nice for the user. Intuitive circuit manipulations can be done with pytket and then sent to qujax for fast implementation, differentiation, parallelisation etc

Reply:
with regards to breaking the API, you are right probably the earlier the better

However, I know `tk_to_jax_symbolic` is currently used by some users so we should be wary

Reply:
If they are currently using this, it should not be a blocker for us. They are not forced to upgrade?